### PR TITLE
KCE-70 Migrate zmq tutorial info from kpe-tutorial to kpsr-tutorial

### PR DESCRIPTION
**Pull request by joseluisbf-kpsr**
[KCE-70]

### TODO

a. ~CI update -> minor update, requires kpsr-core v11 to test~ -> done
b. ~Remove YAML, if possible~ -> done but `kidl/*.yaml`, question below
c. ~Run and verify every example, new Jira?~ -> yes, [KCE-88] to @ThanosAndr 
d. ~Verify cut-and-paste from code to doc, new Jira?~ -> yes, [KCE-88] to @ThanosAndr 

### Questions

1. ~Upgrade the tutorial examples to the latest SDK and core? -> yes, TODO~ -> done
2. ~Remove ZMQ chapter from kpe-tutorial after merge? -> yes, TODO~ -> https://github.com/klepsydra-technologies/kpe-tutorial/pull/21
3. ~Remove `kidl/*.yaml` for `kpsr_codegen`? (In `chapter3_part1.md`)~ -> no, python related --Mandar daily meeting 2023-09-25

[KCE-70]: https://klepsydra.atlassian.net/browse/KCE-70?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[KCE-88]: https://klepsydra.atlassian.net/browse/KCE-88?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[KCE-88]: https://klepsydra.atlassian.net/browse/KCE-88?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: examples/chapter2/zmq_examples/src/environment_zmq.cpp [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    std::string topic = "example2";
+    zmq::context_t context(1);
+    zmq::socket_t publisher(context, ZMQ_PUB);
+    publisher.bind(serverUrl);
+    std::string clientUrl = "tcp://localhost:9001";
+    zmq::socket_t subscriber(context, ZMQ_SUB);
+    subscriber.connect(clientUrl);
+    subscriber.set(zmq::sockopt::subscribe, topic);
+
+    kpsr::zmq_mdlw::ZMQEnv environment("dummy.json", "test", topic, 100, publisher, subscriber);
```
Comment (joseluisbf-kpsr) [inline examples/chapter2/zmq_examples/src/environment_zmq.cpp:33]: Any way to do without a file in this environment API version?

**Comment by pablogh-2000** [comment]
> [KCE-70](https://klepsydra.atlassian.net/browse/KCE-70?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ)
> ### TODO
> 
>     1. CI update
> 
> 
> ### Questions
> 
>     1. Upgrade the tutorial examples to the latest SDK and core?
> 
>     2. Remove ZMQ chapter from kpe-tutorial after merge?

1. Yes
2. Yes
We also need CI for this.

**Review by pablogh-2000 — COMMENTED** [review]
Looks good to me. Do we still need YAML?

**Comment by joseluisbf-kpsr** [comment]
> Do we still need YAML?

No, only for kpsr_codegen, see above.

File: .github/actions/deps/action.yml [hunk 2]
```diff
@@ -4,25 +4,43 @@ runs:
...
+        make install
+      working-directory: kpsr-sdk
+
     - name: Clone kpsr-core
-      uses: actions/checkout@v3
+      uses: actions/checkout@v4
       with:
         repository: klepsydra-technologies/kpsr-core
-        ref: main
+        ref: KCE-85-Update-kpsr-core
```
Comment (joseluisbf-kpsr) [inline .github/actions/deps/action.yml:28]: back to main once merged.

File: .github/actions/deps/action.yml [hunk 3]
```diff
@@ -4,25 +4,43 @@ runs:
   using: "composite"
 
   steps:
+    - name: Clone kpsr-sdk
+      uses: actions/checkout@v4
+      with:
+        repository: klepsydra-technologies/kpsr-sdk
+        ref: KCE-84-Create-kpsr-sdk
```
Comment (joseluisbf-kpsr) [inline .github/actions/deps/action.yml:11]: back to main once merged.

**Review by pablogh-2000 — DISMISSED** [review]

**Comment by pablogh-2000** [comment]
> [KCE-70](https://klepsydra.atlassian.net/browse/KCE-70?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ)
> ### TODO
> 
> a. ~CI update -> minor update, requires kpsr-core v11 to test~ -> done b. ~Remove YAML, if possible~ -> done but `kidl/*.yaml`, question below c. Run and verify every example, new Jira? d. Verify cut-and-paste from code to doc, new Jira?
> ### Questions
> 
>     1. ~Upgrade the tutorial examples to the latest SDK and core? -> yes, TODO~ -> done
> 
>     2. ~Remove ZMQ chapter from kpe-tutorial after merge? -> yes, TODO~ -> [KCE-70 Migrate zmq tutorial info from kpe-tutorial to kpsr-tutorial kpe-tutorial#21](https://github.com/klepsydra-technologies/kpe-tutorial/pull/21)
> 
>     3. ~Remove `kidl/*.yaml` for `kpsr_codegen`? (In `chapter3_part1.md`)~ -> no, python related --Mandar daily meeting 2023-09-25

Please create a new JIRA for points c. and d. and assigned to @ThanosAndr

**Review by pablogh-2000 — APPROVED** [review]

**Comment by joseluisbf-kpsr** [comment]
> Please create a new JIRA for points c. and d. and assigned to @ThanosAndr

[KCE-88] Verify kpsr-tutorial examples

**Pull request merged**